packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -0,0 +1,398 @@
+        bool open;
+        bool executed;
+        ProposalParameters parameters;
+        Tally tally;
+        mapping(address => bool) approvers;
+        IDAO.Action[] actions;
+    }
+
+    /// @notice A container for the proposal parameters.
+    /// @param minApprovals The number of approvals required.
+    /// @param snapshotBlock The number of the block prior to the proposal creation.
+    struct ProposalParameters {
+        uint256 minApprovals;
+        uint64 snapshotBlock;
+    }
+
+    /// @notice A container for the proposal tally.
+    /// @param approvals The number of approvals casted.
+    /// @param addresslistLength The length of the .
+    struct Tally {
+        uint256 approvals;
+        uint256 addresslistLength;
+    }
+
+    /// @notice The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID of the contract.
+    bytes4 internal constant MULTISIG_INTERFACE_ID =
+        this.addAddresses.selector ^
+            this.removeAddresses.selector ^
+            this.isListed.selector ^
+            this.addresslistLength.selector ^
+            this.initialize.selector;
+
+    /// @notice The ID of the permission required to call the `addAddresses` and `removeAddresses` functions.
+    bytes32 public constant UPDATE_MULTISIG_SETTINGS_PERMISSION_ID =
+        keccak256("UPDATE_MULTISIG_SETTINGS_PERMISSION");
+
+    /// @notice The minimum approval parameter.
+    uint256 private minApprovals_;
+
+    /// @notice A mapping between proposal IDs and proposal information.
+    mapping(uint256 => Proposal) internal proposals;
+
+    /// @notice Thrown when a sender is not allowed to create a proposal.
+    /// @param sender The sender address.
+    error ProposalCreationForbidden(address sender);
+
+    /// @notice Thrown if a approver is not allowed to cast an approve. This can be because the proposal
+    /// - is not open,
+    /// - was executed, or
+    /// - the approver is not on the address list
+    /// @param proposalId The ID of the proposal.
+    /// @param sender The address of the sender.
+    error ApprovalCastForbidden(uint256 proposalId, address sender);
+
+    /// @notice Thrown if the proposal execution is forbidden.
+    /// @param proposalId The ID of the proposal.
+    error ProposalExecutionForbidden(uint256 proposalId);
+
+    /// @notice Thrown if the minimal approvals value is out of bounds (less than 1 or greater than the number of members in the address list).
+    /// @param limit The maximal value.
+    /// @param actual The actual value.
+    error MinApprovalsOutOfBounds(uint256 limit, uint256 actual);
+
+    /// @notice Emitted when new members are added to the address list.
+    /// @param members The array of member addresses to be added.
+    event AddressesAdded(address[] members);
+
+    /// @notice Emitted when members are removed from the address list.
+    /// @param members The array of member addresses to be removed.
+    event AddressesRemoved(address[] members);
```

Comment:
Wouldn't these events go to the AddressList base contract/interface?

Reply:
We don't know if the contract inheriting from `Addresslist.sol` wants to emit events.
But yes, I can put them there.
Alternatively, we could but them in a separate interface for `Multisig` and `AddresslistVoting` to inherit from.

Reply:
The reason why this base contract emerges is because we want to share code between Multisig and AddressList, right?
We both know both of them now. If something else is to emerge in the future, we can always move the interface code without breaking the existing contracts.

Reply:
@brickpop I had this in my review, so Michael and I talked and these 2 events are now put in AddressList. So, I'm resolving this .